backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5752730_sys_BusinessRule_M_ProductPrice_C_UOM_Conversion_CatchWeight.sql

```diff
@@ -0,0 +1,110 @@
+;
+
+-- Value: C_UOM_Conversion_CatchWeight_NotExists
+-- 2025-04-24T10:45:00.658Z
+UPDATE AD_Message_Trl SET MsgText='Es gibt noch keine Maßeinheit Umrechnung zu einer Catch-Maßeinheit.',Updated=TO_TIMESTAMP('2025-04-24 10:45:00.658000','YYYY-MM-DD HH24:MI:SS.US')::timestamp without time zone AT TIME ZONE 'UTC',UpdatedBy=100 WHERE AD_Language='de_CH' AND AD_Message_ID=545537
+;
+
+-- 2025-04-24T10:45:00.660Z
+UPDATE AD_Message base SET MsgText=trl.MsgText, Updated=trl.Updated, UpdatedBy=trl.UpdatedBy FROM AD_Message_Trl trl  WHERE trl.AD_Message_ID=base.AD_Message_ID AND trl.AD_Language='de_CH' AND trl.AD_Language=getBaseLanguage()
+;
+
+-- Value: C_UOM_Conversion_CatchWeight_NotExists
+-- 2025-04-24T10:45:02.063Z
+UPDATE AD_Message_Trl SET IsTranslated='Y',Updated=TO_TIMESTAMP('2025-04-24 10:45:02.063000','YYYY-MM-DD HH24:MI:SS.US')::timestamp without time zone AT TIME ZONE 'UTC',UpdatedBy=100 WHERE AD_Language='de_CH' AND AD_Message_ID=545537
+;
+
+-- Value: C_UOM_Conversion_CatchWeight_NotExists
+-- 2025-04-24T10:45:09.170Z
+UPDATE AD_Message_Trl SET MsgText='Es gibt noch keine Maßeinheit Umrechnung zu einer Catch-Maßeinheit.',Updated=TO_TIMESTAMP('2025-04-24 10:45:09.170000','YYYY-MM-DD HH24:MI:SS.US')::timestamp without time zone AT TIME ZONE 'UTC',UpdatedBy=100 WHERE AD_Language='de_DE' AND AD_Message_ID=545537
+;
+
+-- 2025-04-24T10:45:09.171Z
+UPDATE AD_Message base SET MsgText=trl.MsgText, Updated=trl.Updated, UpdatedBy=trl.UpdatedBy FROM AD_Message_Trl trl  WHERE trl.AD_Message_ID=base.AD_Message_ID AND trl.AD_Language='de_DE' AND trl.AD_Language=getBaseLanguage()
+;
+
+-- Value: C_UOM_Conversion_CatchWeight_NotExists
+-- 2025-04-24T10:45:09.272Z
+UPDATE AD_Message_Trl SET IsTranslated='Y',Updated=TO_TIMESTAMP('2025-04-24 10:45:09.272000','YYYY-MM-DD HH24:MI:SS.US')::timestamp without time zone AT TIME ZONE 'UTC',UpdatedBy=100 WHERE AD_Language='de_DE' AND AD_Message_ID=545537
+;
+
+-- Name: M_ProductPrice Catchweight has UOM Conversion
+-- 2025-04-24T10:51:46.434Z
+INSERT INTO AD_Val_Rule (AD_Client_ID,AD_Org_ID,AD_Val_Rule_ID,Created,CreatedBy,EntityType,IsActive,Name,Type,Updated,UpdatedBy) VALUES (0,0,540716,TO_TIMESTAMP('2025-04-24 10:51:46.431000','YYYY-MM-DD HH24:MI:SS.US')::timestamp without time zone AT TIME ZONE 'UTC',100,'U','Y','M_ProductPrice Catchweight has UOM Conversion','S',TO_TIMESTAMP('2025-04-24 10:51:46.431000','YYYY-MM-DD HH24:MI:SS.US')::timestamp without time zone AT TIME ZONE 'UTC',100)
+;
+
+-- Name: M_ProductPrice Catchweight has UOM Conversion
+-- 2025-04-24T10:51:49.692Z
+UPDATE AD_Val_Rule SET Code='EXISTS (
+    SELECT 1
+    FROM M_ProductPrice pp
+             JOIN M_Product p
+                  ON pp.M_Product_ID = p.M_Product_ID
+    WHERE pp.M_ProductPrice_ID = M_ProductPrice.M_ProductPrice_ID
```

Comment:
why are u selecting again from `M_ProductPrice pp`  when u already have the M_ProductPrice record available?!